app/controllers/orders_controller.rb

```diff
@@ -28,8 +25,9 @@ def edit
   # POST /orders
   # POST /orders.json
   def create
-    if current_dispatcher.present?
-      @order = Order.new(order_params)
+    @dispatcher = Dispatcher.find(params[:dispatcher_id])
```

Comment:
From where do you get `dispatcher_id`? To get id of a current dispatcher (if dispatcher is present), use `current_dispatcher.id`.